File: rest_api/php_samples/package_creation_token.php
<?php
$token = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"; // replace this with your token key
$url = "http://test.delhivery.com/cmu/push/json/?token=".$token;

$params = array(); // this will contain request meta and the package feed
$package_data = array(); // package data feed
$shipments = array();
$pickup_location = array();

/////////////start: building the package feed/////////////////////
$shipment = array();
$shipment['name'] = 'John Kapoor'; // consignee name
$shipment['order'] = '3002199823224D'; // client order number
$shipment['products_desc'] = 'Resume services';
$shipment['order_date'] = '2013-04-08T18:30:00+00:00'; // ISO Format
$shipment['payment_mode'] = 'Cash';
$shipment['total_amount'] = 21841.0; // in INR
$shipment['cod_amount'] = '21841.0'; // amount to be collected, required for COD
$shipment['add'] = 'M25, Nelson Marg, GBP City Phase 1'; // consignee address
$shipment['city'] = 'Gurgaon';
$shipment['state'] = 'Haryana';
$shipment['country'] = 'India';
$shipment['phone'] = '9741119727';
$shipment['pin'] = '122002';

// pickup location information //
$pickup_location['add'] = '113 Barthal, Dundahera';
$pickup_location['city'] = 'New Delhi';
$pickup_location['country'] ='India';
$pickup_location['name'] = 'client_location';  // Use client warehouse name
$pickup_location['phone'] = '011-23456245';
$pickup_location['pin'] = '110070';
$pickup_location['state'] = 'Delhi';

$shipments = array($shipment);

$package_data['shipments'] = $shipments;
$package_data['pickup_location'] = $pickup_location;
$params['format'] = 'json';
$params['data'] =json_encode($package_data);

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS,http_build_query($params));
curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/x-www-form-urlencoded'));
$result = curl_exec($ch);
print_r($result);
curl_close($ch);

?>

File: rest_api/python_samples/package_creation_token.py
import requests, simplejson, urllib
import random


shipments= [{
	"waybill":"1111111111", # waybill number
        "name": "Rohan Gothwal", 
        "order": "32423424", # order number unique
        "products_desc": "Sony PS3 Super Slim 500 GB (Black)",
        "order_date": "2014-05-08T12:30:00+00:00",
        "payment_mode": "Pre-paid",
        "total_amount": 21841.0,
        "cod_amount": 0.0,
        "add": "M25, Nelson Marg, GBP City Phase 1",
        "city": "Gurgaon",
        "state": "Haryana",
        "country": "India",
        "phone": "9741119727",
        "pin": "122002",
        "return_add": "",
        "return_city": "",
        "return_country": "",
        "return_name": "",
        "return_phone": "",
        "return_pin": "",
        "return_state": "",
        "supplier": "Kangaroo (India) Pvt Ltd",
        "billable_weight": 650.0,
        "dimensions": "0.00CM x 0.00CM x 0.00CM",
        "volumetric": 0.0,
        "weight": "650.0 gm",
	"product_quantity": 1, # quantity of goods, positive integer
	"seller_inv": "invoice number",
	"seller_inv_date": 'YYYY-MM-DDTHH:MM:SS+05:30', # ISO format
	"seller_name": "seller name", # name of seller
	"seller_add": "seller add", # seller address
	"seller_cst": "seller cst", # seller cst no
	"seller_tin": "seller tin", # seller tin no

}
        ]
token="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
url = "http://test.delhivery.com/cmu/push/json/?token={}".format(token)
r = requests.post(url, data=urllib.urlencode(
    {
        "data":simplejson.dumps(
            {
                "dispatch_date": "2014-05-08T18:00:00.000000+05:30",
                "shipments": shipments,
            }
        ),
        "format": "json"
    }), headers = {"content-type": "application/x-www-form-urlencoded"})
print r

